backend/src/main/java/com/google/sps/FindMatchQuery.java

```diff
@@ -36,23 +37,29 @@ public FindMatchQuery(Clock clock) {
   }
 
   /**
-   * Find match of new participant or add to list of participants, return whether or not match was
-   * found right after being added
+   * Find and return match of new participant with unmatched participants by comparing duration and
+   * availibility or return null if no match yet
    */
-  public Match findMatch(List<Participant> participants, Participant newParticipant) {
+  public Match findMatch(ParticipantDatastore participantDatastore, Participant newParticipant) {
+    // Get list of unmatched participants in datastore
+    List<Participant> unmatchedParticipants = participantDatastore.getUnmatchedParticipants();
+
+    // Add newParticipant to datastore
+    participantDatastore.addParticipant(newParticipant);
```

Comment:
This seems like an unexpected side effect of this method. Could it be pushed up to the callers? Should a participant who is matched by this algorithm be inserted into the datastore at all?

Reply:
Moved it to AddParticipantServlet. I put it into the match method for the tests but I found an easy way to test still.

Reply:
Also, I changed it so that all participants, once inserted into the datastore, stay forever. Their inputs are just nulled out if a match has been found or their time is up.